native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/QueryProcessor.java

```diff
@@ -66,61 +74,113 @@ private QueryProcessor() {
     public static BStream nativeQuery(
             BObject client, Object paramSQLString,
             Object recordType,
+            AbstractStatementParameterProcessor statementParameterProcessor,
+                AbstractResultParameterProcessor resultParameterProcessor) {
+        ResultSet resultSet = null;
+        try {
+            resultSet = initQuery(client, paramSQLString, statementParameterProcessor);
+            RecordType streamConstraint = (RecordType) ((BTypedesc) recordType).getDescribingType();
+            List<ColumnDefinition> columnDefinitions = Utils.getColumnDefinitions(resultSet, streamConstraint);
+            return ValueCreator.createStreamValue(TypeCreator.createStreamType(streamConstraint,
+                    PredefinedTypes.TYPE_NULL), resultParameterProcessor
+                    .createRecordIterator(resultSet, statement, connection, columnDefinitions, streamConstraint));
+        } catch (SQLException e) {
+            Utils.closeResources(trxResourceManager, resultSet, statement, connection);
+            BError errorValue = ErrorGenerator.getSQLDatabaseError(e,
+                    "Error while executing SQL query: " + sqlQuery + ". ");
+            return ValueCreator.createStreamValue(TypeCreator.createStreamType(Utils.getDefaultStreamConstraint(),
+                    PredefinedTypes.TYPE_NULL), createRecordIterator(errorValue));
+        } catch (ApplicationError applicationError) {
+            Utils.closeResources(trxResourceManager, resultSet, statement, connection);
+            BError errorValue = ErrorGenerator.getSQLApplicationError(applicationError.getMessage());
+            return getErrorStream(recordType, errorValue);
+        } catch (Throwable e) {
+            Utils.closeResources(trxResourceManager, resultSet, statement, connection);
+            String message = e.getMessage();
+            if (message == null) {
+                message = e.getClass().getName();
+            }
+            BError errorValue = ErrorGenerator.getSQLApplicationError(
+                    "Error while executing SQL query: " + sqlQuery + ". " + message);
+            return getErrorStream(recordType, errorValue);
+        }
+    }
+
+    public static Object nativeQueryRow(
+            BObject client, Object paramSQLString,
+            BTypedesc ballerinaType,
             AbstractStatementParameterProcessor statementParameterProcessor,
             AbstractResultParameterProcessor resultParameterProcessor) {
+        ResultSet resultSet = null;
+        try {
+            resultSet = initQuery(client, paramSQLString, statementParameterProcessor);
+
+            if (!resultSet.next()) {
```

Comment:
Just thinking on this, how would this behave if a user expects nil? For in stance I can write a query with an on condition which can either return a record or nil
```
record {}? value = client->queryRow()
```
How will the implementation behave in this case?

Reply:
Currently it will return an error if the query retrieves an empty result set. Can configure to return null if nil is an acceptable return type.

Reply:
We have decided to Go with returning NoRowsError